homeassistant/components/esphome/voice_assistant.py

```diff
@@ -108,6 +109,22 @@ def accept_connection() -> VoiceAssistantUDPServer:
         sock = socket.socket(socket.AF_INET, socket.SOCK_DGRAM)
         sock.setblocking(False)
 
+        def find_port(port: int, port_max: int) -> int:
+            sock = socket.socket(socket.AF_INET, socket.SOCK_DGRAM)
+            while port <= port_max:
+                try:
+                    sock.bind(("", port))
+                    sock.close()
+                    return port
+                except OSError:
+                    port += 1
+            raise OSError("No free ports")
+
+        if UDP_PORT_MIN == 0:
+            UDP_PORT = 0
+        else:
+            UDP_PORT = find_port(UDP_PORT_MIN, UDP_PORT_MAX)
+
         sock.bind(("", UDP_PORT))
```

Comment:
The variable shouldn't be caps if it is local
```suggestion
            udp_port = 0
        else:
            udp_port = find_port(UDP_PORT_MIN, UDP_PORT_MAX)

        sock.bind(("", udp_port))
```